vue-components/src/components/TextInput.vue

```diff
@@ -0,0 +1,202 @@
+<template>
+	<label :class="classes">
+		<div class="wikit-TextInput__label">{{ label }}</div>
+		<input
+			type="text"
+			class="wikit-TextInput__input"
+			:value="value"
+			:placeholder="placeholder"
+			:disabled="disabled"
+		>
+		<div class="wikit-TextInput__error-message" v-if="error">
+			<Icon
+				class="wikit-TextInput__error-message__icon"
+				:type="error.type === 'error' ? 'error' : 'alert'"
+				:color="error.type"
+				size="medium"
+			/>
+			{{ error.message }}
+		</div>
+	</label>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import Icon from './Icon.vue';
+
```

Comment:
/* eslint-disable no-trailing-spaces */
/**
 *  Text input fields are form elements that let users input and edit values in the form of text.
 */
/* eslint-enable no-trailing-spaces */
```suggestion

```